### ISSUE TITLE
CockroachDB: error while creating a table with date type columns and default value NOW()

### ISSUE THREAD

**Issue by lucarestagno** [hasura/graphql-engine#9011]
{
    "arguments": [],
    "error": {
        "description": "source SQL:\nCREATE OR REPLACE FUNCTION \"public\".\"set_current_timestamp_updatedAt\"()\nRETURNS TRIGGER AS $$\nDECLARE\n  _new record;\nBEGIN\n  _new := NEW;\n  _new.\"updatedAt\" = NOW();\n  RETURN _new;\nEND;\n$$ LANGUAGE plpgsql\n            ^",
        "exec_status": "FatalError",
        "hint": null,
        "message": "at or near \"plpgsql\": syntax error: language \"plpgsql\" does not exist",
        "status_code": "42601"
    },
    "prepared": false,
    "statement": "CREATE TABLE \"public\".\"Users\" (\"id\" serial NOT NULL, \"firstName\" text NOT NULL, \"lastName\" text NOT NULL, \"createdAt\" timestamptz NOT NULL DEFAULT now(), \"updatedAt\" timestamptz NOT NULL DEFAULT now(), PRIMARY KEY (\"id\") );\nCREATE OR REPLACE FUNCTION \"public\".\"set_current_timestamp_updatedAt\"()\nRETURNS TRIGGER AS $$\nDECLARE\n  _new record;\nBEGIN\n  _new := NEW;\n  _new.\"updatedAt\" = NOW();\n  RETURN _new;\nEND;\n$$ LANGUAGE plpgsql;\nCREATE TRIGGER \"set_public_Users_updatedAt\"\nBEFORE UPDATE ON \"public\".\"Users\"\nFOR EACH ROW\nEXECUTE PROCEDURE \"public\".\"set_current_timestamp_updatedAt\"();\nCOMMENT ON TRIGGER \"set_public_Users_updatedAt\" ON \"public\".\"Users\" \nIS 'trigger to set value of column \"updatedAt\" to current timestamp on row update';"
}

**Comment by danieljharvey** [comment]
This will be failing because Cockroach DB does not support user defined functions, and our `updated_at` code tries to add a trigger. Probably simplest to remove this suggested field for now.

It seems to live in `frontend/libs/console/legacy-oss/src/lib/dataSources/services/cockroach/sqlUtils.ts`

**Comment by lucarestagno** [comment]
> This will be failing because Cockroach DB does not support user defined functions, and our `updated_at` code tries to add a trigger. Probably simplest to remove this suggested field for now.
> 
> It seems to live in `frontend/libs/console/legacy-oss/src/lib/dataSources/services/cockroach/sqlUtils.ts`

I agree, we could just remove the suggested column "updated_at" from the list for Cockroach.
@dameleney what do you think?

**Comment by ajohnson1200** [comment]
> This will be failing because Cockroach DB does not support user defined functions

Maybe I'm not reading the documentation the correct way, but it seems like "now()" is supported:

https://www.cockroachlabs.com/docs/stable/functions-and-operators.html#date-and-time-functions

no?

**Comment by danieljharvey** [comment]
Ahh yes, it is, we could use `NOW()` to make a `created_at` volumn with something like `ADD COLUMN created_at TIMESTAMP DEFAULT NOW()`, however we can't make an auto updating `updated_at` column without functions.

This is what it's currently doing:
https://github.com/hasura/graphql-engine-mono/blob/main/frontend/libs/console/legacy-oss/src/lib/dataSources/services/cockroach/sqlUtils.ts#L1165

**Comment by sassela** [comment]
@danieljharvey did your PR get merged, and did it resolve the issue? If so, please could you link the PR and close this?

**Comment by lucarestagno** [comment]
issue solved with https://github.com/hasura/graphql-engine-mono/pull/6070

we removed `updated_at` form the suggested fields for CockroachDB only.
